wgpu-core/src/device/queue.rs

```diff
@@ -1062,7 +1062,7 @@ impl Global {
             let fence = fence_guard.as_mut().unwrap();
             let submit_index = device
                 .active_submission_index
-                .fetch_add(1, Ordering::Relaxed)
+                .fetch_add(1, Ordering::SeqCst)
```

Comment:
I still don't think this is any more correct, but not with thinking about the cost, so we can leave the change

Reply:
You're right, we need to make sure that submissions occur in order of increasing index, which means we need a lock on this entire thing. Using `SeqCst` here isn't good enough.

Reply:
Filed as #5978